core/src/consensus.rs

```diff
@@ -564,29 +581,35 @@ impl Tower {
         trace!("{} record_vote for {}", self.node_pubkey, vote_slot);
         let old_root = self.root();
 
-        let mut new_vote = if is_direct_vote_state_update_enabled {
+        if is_direct_vote_state_update_enabled {
             let vote = Vote::new(vec![vote_slot], vote_hash);
-            process_vote_unchecked(&mut self.vote_state, vote);
-            VoteTransaction::from(VoteStateUpdate::new(
-                self.vote_state
-                    .votes
-                    .iter()
-                    .map(|vote| vote.lockout)
-                    .collect(),
-                self.vote_state.root_slot,
-                vote_hash,
-            ))
+            let epoch = self.vote_state.current_epoch();
+            let result = process_vote_unfiltered(
+                &mut self.vote_state,
+                &vote.slots,
+                &vote,
+                &[(vote_slot, vote_hash)],
+                epoch,
+            );
+            if result.is_err() {
```

Comment:
maybe just bubble up the error from `process_vote_unchecked` and error log here?